### PR TITLE
LCO/NPV fix

### PR DESCRIPTION
**Pull request by jaredthomas68**
# LCO/NPV fix (pad with non-zero values)

@vijay092 noticed that running ProFASTNPV with the price found in ProFASTLCO did not always result in a zero NPV as expected. The reason was the padding values in the price are sometimes used for cases where installation months is not divisible by 12. See #874 for detailed discussion.

This PR includes the fix proposed by @vijay092 along with corresponding tests to ensure proper behavior in the future.

## Section 1: Type of Contribution

- [ ] Feature Enhancement
  - [ ] Framework
  - [ ] New Model
  - [ ] Updated Model
  - [ ] Tools/Utilities
  - [ ] Other (please describe):
- [x] Bug Fix
- [ ] Documentation Update
- [ ] CI Changes
- [ ] Other (please describe):

## Section 2: Draft PR Checklist

- [ ] Open draft PR
- [ ] Describe the feature that will be added
- [ ] Fill out TODO list steps
- [ ] Describe requested feedback from reviewers on draft PR
- [ ] Complete Section 8: New Model Checklist (if applicable)

### TODO:
- [ ] Step 1
- [ ] Step 2

### Type of Reviewer Feedback Requested (on Draft PR)

**Structural feedback**:

**Implementation feedback**:

**Other feedback**:

## Section 3: General PR Checklist

- [x] PR description thoroughly describes the new feature, bug fix, etc.
- [x] Added tests for new functionality or bug fixes
- [x] Tests pass (If not, and this is expected, please elaborate in the Section 6: Test Results)
- [-] Documentation
  - [-] Docstrings are up-to-date
  - [-] Related `docs/` files are up-to-date, or added when necessary
  - [-] Documentation has been rebuilt successfully
  - [-] Examples have been updated (if applicable)
- [x] `CHANGELOG.md`
  - [x] At least one complete sentence has been provided to describe the changes made in this PR
  - [x] After the above, a hyperlink has been provided to the PR using the following format:
    "A complete thought. [PR XYZ]((https://github.com/NatLabRockies/H2Integrate/pull/XYZ)", where
    `XYZ` should be replaced with the actual number.

## Section 4: Related Issues

#874 

### Section 5.1: Renamed Files
- test_profast_finance -> test_profast_lco.py
  - Renamed to test_profast_lco.py to reflect actual test scope (LCO-specific).

### Section 5.2: Modified Files
- profast_npv.py
  - Updated construction-period price padding to use the first operating-year sell price instead of zeros so NPV-at-LCOE behavior is consistent with ProFAST timing assumptions.
- test_profast_npv.py
  - Added and simplified a regression test that verifies the padded price profile passed to ProFAST and guards against reintroducing zero-padding behavior.
- test_finances.py
  - Added an H2IntegrateModel integration test that computes LCOE, feeds it into ProFastNPV sell price, reruns, and checks NPV is approximately zero.

## Section 8 (Optional): New Model Checklist

- [ ] **Model Structure**:
  - [ ] Follows established naming conventions outlined in `docs/developer_guide/coding_guidelines.md`
  - [ ] Used `attrs` class to define the `Config` to load in attributes for the model
    - [ ] If applicable: inherit from `BaseConfig` or `CostModelBaseConfig`
  - [ ] Added: `initialize()` method, `setup()` method, `compute()` method
    - [ ] If applicable: inherit from `CostModelBaseClass`
- [ ] **Integration**: Model has been properly integrated into H2Integrate
  - [ ] Add the new model to the appropriate `__init__.py` file to ensure it is properly imported and used in `supported_models.py`
  - [ ] Added to `supported_models.py`
  - [ ] If a new commodity_type is added, update `create_financial_model` in `h2integrate_model.py`
- [ ] **Tests**: Unit tests have been added for the new model
  - [ ] [Pytest-style unit tests](https://realpython.com/pytest-python-testing/)
  - [ ] Unit tests are in a "test" folder within the folder a new model was added to
  - [ ] If applicable add integration tests
- [ ] **Example**: If applicable, a working example demonstrating the new model has been created
  - [ ] Input file comments
  - [ ] Run file comments
  - [ ] Example has been tested and runs successfully in `test_all_examples.py`
- [ ] **Documentation**:
  - [ ] Write docstrings using the [Google style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)
  - [ ] Model added to the main models list in `docs/user_guide/model_overview.md`
    - [ ] Model documentation page added to the appropriate `docs/` section
    - [ ] `<model_name>.md` is added to the `_toc.yml`
  - [ ] Run `generate_class_hierarchy.py` to update the class hierarchy diagram in `docs/developer_guide/class_structure.md`

### CODE REVIEW

File: h2integrate/finances/test/test_profast_lco.py
Comment (elenya-grant) [inline h2integrate/finances/test/test_profast_lco.py:1]: thanks for renaming this - long overdue!

File: h2integrate/finances/test/test_finances.py [hunk 1]
```diff
@@ -221,6 +227,125 @@ def test_lcoe_with_selected_technologies():
...
+@pytest.mark.integration
+def test_lcoe_price_drives_profast_npv_to_zero():
+    n_timesteps = 8760
+    plant_life = 30
+    grid_setpoint = 40000.0  # kW
+
+    shared_params = {
+        "analysis_start_year": 2032,
+        "installation_time": 14,
+        "inflation_rate": 0.0,
```
Comment (elenya-grant) [inline h2integrate/finances/test/test_finances.py:239]: I'm thinking it could be good to also have a test with `inflation_rate` as non-zero but thats a nit-pick! Thanks for adding this test in!

**Review by elenya-grant — APPROVED** [review]
thanks for this PR Jared! Just to double-check, this bug is specific to the NPV model, not the LCO model - right? Also, thank you for figuring out the exact reason for this bug, it's nice to know that it's because of the non-divisible-by-12 installation period! Assuming all tests pass - this looks good to me!

File: h2integrate/finances/test/test_finances.py (hunk 1 above)
Comment (jaredthomas68) [reply h2integrate/finances/test/test_finances.py:239]: Done
